src/token/src/types.rs

```diff
@@ -1,12 +1,40 @@
 use candid::{CandidType, Deserialize, Nat, Principal};
-use common::types::Metadata;
-use std::collections::{HashMap, HashSet};
+use ic_helpers::is20::TxError;
+use std::collections::HashMap;
 
 mod tx_record;
 pub use tx_record::*;
 
 pub type Timestamp = u64;
 
+#[allow(non_snake_case)]
+#[derive(Deserialize, CandidType, Clone, Debug)]
+pub struct Metadata {
+    pub logo: String,
+    pub name: String,
+    pub symbol: String,
+    pub decimals: u8,
+    pub totalSupply: Nat,
+    pub owner: Principal,
+    pub fee: Nat,
+    pub feeTo: Principal,
+    pub isTestToken: Option<bool>,
+}
+
+#[derive(CandidType, Debug, Clone, Deserialize)]
+pub struct SignedTx {
+    /// Principal of token that called `receive_is20`
+    pub principal: Principal,
+    /// Pubkey associated to a principal.
+    pub publickey: Vec<u8>,
+    /// Transaction signing signature on behalf of `publickey`.
+    pub signature: Vec<u8>,
+    /// Transaction serialized with `serde-cbor`.
+    pub serialized_tx: Vec<u8>,
```

Comment:
Do we want to return serialized transaction here though? It would be more convinient for the consumer to have the transaction itself. If the consumer wants to check the signature, he can serialize the value (serialization is usually more easy and less error prone then deserialization).